core/shared/src/main/scala/zio/VersionSpecific.scala

```diff
@@ -57,7 +57,7 @@ private[zio] trait VersionSpecific {
   type LightTypeTag = izumi.reflect.macrortti.LightTypeTag
 
   private[zio] def taggedIsSubtype(left: LightTypeTag, right: LightTypeTag): Boolean =
-    left <:< right
+    taggedSubtypes.getOrElseUpdate((left, right), left <:< right)
```

Comment:
This is not concurrent-safe. Maybe concurrent map?

Reply:
Yes. I don't think it is an actual issue if there are concurrent updates as we will just compute the value an additional time but there is a risk that values memoized by one thread might never be visible on other threads.

Reply:
I think it's a mutable map which is not guaranteed to not throw exceptions in the presence of concurrent updates.

Reply:
Doesn't look like it throws exceptions on concurrent updates though maybe that is an implementation detail we can't rely on.